ably/options.go

```diff
@@ -450,37 +474,62 @@ func (opts *clientOptions) activePort() (port int, isDefault bool) {
 	return
 }
 
-func (opts *clientOptions) getRestHost() string {
+// endpointFqdn handles an endpoint that uses a hostname, which may be an IPv4
+// address, IPv6 address or localhost
+func endpointFqdn(endpoint string) bool {
+	return strings.Contains(endpoint, ".") || strings.Contains(endpoint, "::") || endpoint == "localhost"
+}
+
+func (opts *clientOptions) endpointValueWithLegacySupport() string {
+	if !empty(opts.Endpoint) {
+		return opts.Endpoint
+	}
+
+	if !empty(opts.Environment) {
+		if opts.Environment == "production" {
+			return defaultOptions.Endpoint
+		}
+
+		return opts.Environment
+	}
+
 	if !empty(opts.RESTHost) {
 		return opts.RESTHost
 	}
-	if !opts.isProductionEnvironment() {
-		return opts.Environment + "-" + defaultOptions.RESTHost
-	}
-	return defaultOptions.RESTHost
-}
 
-func (opts *clientOptions) getRealtimeHost() string {
 	if !empty(opts.RealtimeHost) {
 		return opts.RealtimeHost
 	}
-	if !empty(opts.RESTHost) {
-		logger := opts.LogHandler
-		logger.Printf(LogWarning, "restHost is set to %s but realtimeHost is not set so setting realtimeHost to %s too. If this is not what you want, please set realtimeHost explicitly.", opts.RESTHost, opts.RealtimeHost)
-		return opts.RESTHost
+
+	return defaultOptions.Endpoint
+}
+
+// REC2
+func (opts *clientOptions) getEndpoint() string {
```

Comment:
Given this is taking the configured endpoint, like `main`, and turning that into a hostname, like `main.realtime.ably.net`, perhaps this should be `getHostname()`?

Reply:
90c9a2bcf57260c53ea42f5d93cbc688c3c493d7